sunpy/map/mapbase.py

```diff
@@ -650,10 +704,47 @@ def rotate(self, angle=None, rmatrix=None, order=4, scale=1.0,
             raise ValueError("You cannot specify both an angle and a matrix")
         elif angle is None and rmatrix is None:
             rmatrix = self.rotation_matrix
+        
+        # This is out of the quantity_input decorator. To allow the angle=None 
+        # case. See https://github.com/astropy/astropy/issues/3734
+        if angle:
+            try:
+                equivalent = angle.unit.is_equivalent(u.deg)
+
+                if not equivalent:
+                    raise u.UnitsError("Argument '{0}' to function '{1}'"
+                                       " must be in units convertable to"
+                                       " '{2}'.".format('angle', 'rotate',
+                                                      u.deg.to_string()))
+
+            # Either there is no .unit or no .is_equivalent
+            except AttributeError:
+                if hasattr(angle, "unit"):
+                    error_msg = "a 'unit' attribute without an 'is_equivalent' method"
+                else:
+                    error_msg = "no 'unit' attribute"
+                raise TypeError("Argument '{0}' to function '{1}' has {2}. "
+                      "You may want to pass in an astropy Quantity instead."
+                         .format('angle', 'rotate', error_msg))
 
         # Interpolation parameter sanity
         if order not in range(6):
             raise ValueError("Order must be between 0 and 5")
+        
+        # is None to prevent 0 arrays evaling to False
+        if rotation_center is None:
+            rotation_center = u.Quantity([0*self.units['x'], 0*self.units['y']])
+        else:
+            if not isinstance(rotation_center, u.Quantity) and not hasattr(rotation_center, "unit"):
```

Comment:
Not explicitly, but I am not sure that is a pattern we should support everywhere. That would be like supporting a tuple of numpy arrays, which would be a little odd.